roxie/udplib/udptrr.cpp

```diff
@@ -1533,7 +1534,9 @@ class CReceiveManager : implements IReceiveManager, public CInterface
                 msgColl.set(collators[pktHdr->ruid]);
                 if (!msgColl)
                 {
-                    msgColl.set(collators[RUID_DISCARD]);
+                    // We only send single-packet messages to the default collator
+                    if (pktHdr->pktSeq & UDP_PACKET_COMPLETE)
```

Comment:
I think this might need to also check that the sequence number is 0, otherwise the packet will stay in the package sequencer never be removed.

A couple of other improvements:
- CMessageCollator::collate could be simplified if ruid < RUID_FIRST.  Assert that the packet is complete, and avoid adding it to the mapping table).

Reply:
I agree with the ruid < RUID_FIRST comment.